File: backend/program/model/dto/schedule.go
package dto

type ScheduleDTO struct {
	ID        uint `gorm:"primary_key"`
	ProgramID uint `gorm:"program_id"`
	ChannelID uint `gorm:"channel_id"`
	StartTime uint `gorm:"start_time"`
	EndTime   uint `gorm:"end_time"`
}

File: frontend/src/api/modules/schedule.ts
import axios from 'axios'

export interface Schedule {
  ID: number
  ProgramID: number
  ChannelID: number
  StartTime: number
  EndTime: number
}

export interface ScheduleInfoData {
  schedule_id: number
}

export interface ScheduleInfoRes {
  info: Schedule
}

export interface ScheduleListRes {
  list: number[]
}

export function ScheduleInfo(data: ScheduleInfoData) {
  return axios.post<ScheduleInfoRes>('/api/program/scheduleInfo', data)
}

export function ScheduleList() {
  return axios.get<ScheduleListRes>('/api/program/scheduleList')
}
